api/server.go

```diff
@@ -38,12 +65,123 @@ func (s *Server) walletBalanceHandler(w http.ResponseWriter, req *http.Request,
 	})
 }
 
+func (s *Server) walletAddressHandler(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	writeJSON(w, WalletAddress{s.w.NextAddress()})
+}
+
+func (s *Server) walletAddressesHandler(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	writeJSON(w, WalletAddresses{s.w.Addresses()})
+}
+
+func (s *Server) walletTransactionsHandler(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	writeJSON(w, Transactions{s.w.Transactions()})
+}
+
+func (s *Server) walletSignHandler(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	var txn types.Transaction
+	if err := json.NewDecoder(req.Body).Decode(&txn); err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	var amount types.Currency
+	for _, out := range txn.SiacoinOutputs {
+		amount = amount.Add(out.Value)
+	}
+
+	cm := s.s.ChainManager()
+	vc, err := cm.ValidationContext(cm.Tip())
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	if toSign, _, err := s.w.FundTransaction(&txn, amount, nil); err != nil {
```

Comment:
an endpoint named "sign" should not also be funding the transaction. In `walrus`, the client requests the wallet's unspent outputs, and uses them to construct the transaction locally. We should take the same approach here. We can add a helper endpoint like `/wallet/fund` later if needed.